reconcile/utils/vault.py

```diff
@@ -189,6 +193,13 @@ def read_all_with_version(self, secret: Mapping) -> tuple[Mapping, Optional[str]
         if data is None:
             raise SecretNotFound
 
+        # Check the keys to ensure that they do not have any whitespace around them.
+        for k in data:
+            if k.strip() != k:
+                raise SecretFormatProblem(
```

Comment:
We should add this exception in `retry` decorator through `no_retry_exceptions` to avoid unnecessary extra calls.

Reply:
Done.